tensorflow/c/tf_tensor.h

```diff
@@ -45,6 +45,16 @@ limitations under the License.
 extern "C" {
 #endif
 
+// Allocator Attributes used for tensor allocation 
+struct TF_AllocatorAttributes { 
```

Comment:
```suggestion
typedef struct TF_AllocatorAttributes { 
```

Reply:
Hi, how will `TF_AllocatorAttributes` be used? Maybe you can point me to the API that consumes `TF_AllocatorAttributes`.

Reply:
I will be adding an additional argument of type, TF_AllocatorAttributes, to [allocate_temp ](https://github.com/tensorflow/tensorflow/blob/a4bb80f41efb5d60f404810d6576f47602be77eb/tensorflow/c/kernels.cc#L277). An argument of type AllocatorAttributes will be created from TF_AllocatorAttributes to pass as a parameter to [allocate_temp](https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/framework/op_kernel.h;l=999;drc=356121e56376d4c872848a710fefa4dc3e1f3b6a)

Reply:
Thanks for the explanation. Makes sense for now. 
While out of scope, I think it might be worthwhile to mention that a `typedef struct TF_AllocateTempParams { struct_size ...}` could subsume all of TF_AllocateTemp's input arguments including TF_AllocatorAttributes.

Reply:
I will bring up this comment to my host and see if this is the best course of action. Will submit another PR if so. Thanks!

Reply:
Just something to think about in case there's time for that in future. 99% sure it is not needed right now :)